packages/ui/src/components/SelectInputV2/Dropdown.tsx

```diff
@@ -0,0 +1,916 @@
+                : option.value.toLocaleLowerCase().startsWith(currentSearch),
+            )
+
+            return null
+          })
+          const closestOption =
+            closestOptions[Object.keys(closestOptions)[0]][0]
+          if (closestOption) {
+            setDefaultSearch(closestOption.searchText ?? closestOption.value)
+          } else {
+            setDefaultSearch(null)
+          }
+        } else {
+          const closestOption = [...options].filter(option =>
+            option.searchText
+              ? option.searchText.toLocaleLowerCase().startsWith(currentSearch)
+              : option.value.toLocaleLowerCase().startsWith(currentSearch),
+          )[0]
+          if (closestOption) {
+            setDefaultSearch(closestOption.searchText ?? closestOption.value)
+          } else {
+            setDefaultSearch(null)
+          }
+        }
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
+    document.addEventListener('keydown', handleKeyDown)
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+      document.removeEventListener('keydown', handleKeyDown)
+    }
+  }, [
+    setIsDropdownVisibile,
+    isDropdownVisible,
+    searchBarActive,
+    options,
+    onSearch,
+    search,
+    refSelect,
+    setDefaultSearch,
+  ])
+
+  return ref
+}
+
+type CreateDropdownProps = {
+  displayedOptions: DataType
+  options: DataType
+  isEmpty: boolean
+  emptyState: ReactNode
+  multiselect: boolean
+  descriptionDirection?: 'row' | 'column'
+  selectedValues: OptionType[]
+  setIsDropdownVisible: Dispatch<SetStateAction<boolean>>
+  onChange?: (value: string[]) => void
+  setSelectedValues: Dispatch<SetStateAction<OptionType[]>>
+  loadMore?: ReactNode
+  optionalInfoPlacement: 'left' | 'right'
+  defaultSearchValue: string | null
+  isLoading?: boolean
+  selectAll?: { label: ReactNode; description?: string }
+  allSelected: boolean
+  setAllSelected: Dispatch<SetStateAction<boolean>>
+  selectAllGroup?: boolean
+  selectedGroups: string[]
+  setSelectedGroups: Dispatch<SetStateAction<string[]>>
+  numberOfOptions: number
+}
+const CreateDropdown = ({
+  displayedOptions,
+  isEmpty,
+  emptyState,
+  descriptionDirection,
+  selectedValues,
+  setIsDropdownVisible,
+  onChange,
+  setSelectedValues,
+  loadMore,
+  optionalInfoPlacement,
+  defaultSearchValue,
+  isLoading,
+  selectAll,
+  multiselect,
+  options,
+  allSelected,
+  setAllSelected,
+  selectAllGroup,
+  selectedGroups,
+  setSelectedGroups,
+  numberOfOptions,
+}: CreateDropdownProps) => {
+  const focusedItemRef = useRef<HTMLButtonElement>(null)
+
+  useEffect(() => {
+    if (defaultSearchValue && focusedItemRef?.current) {
+      focusedItemRef.current.focus()
+    }
+  }, [defaultSearchValue])
+
+  if (isEmpty) {
+    return (
+      <EmptyState gap={2} alignItems="center">
+        {emptyState ?? (
+          <Text variant="bodyStrong" as="p">
+            No options
```

Comment:
Text is not customizable ?

Reply:
When you don't pass an emptyState, it will have a default non-customizable text yes